### PR TITLE
felix/bpf: good to know which routes are tunneled

### PR DESCRIPTION
**Pull request by tomastigera**
Internal routes carry the knowledge whether they go via an overlay
tunnel or not. BPF routes where missing this knowledge.

In the current setup, we assume that when Wireguard is used, any pod is
reached by a tunnel without that being explicit in the internal routes.

This is a prerequisite for being able to forward correctly without CTLB
and to do tunneling in BPF entirely.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/bpf-gpl/bpf.h [hunk 1]
```diff
@@ -261,7 +261,7 @@ CALI_CONFIGURABLE_DEFINE(ext_to_svc_mark, 0x4b52414d) /*be 0x4b52414d = ASCII(MA
 CALI_CONFIGURABLE_DEFINE(psnat_start, 0x53545250) /* be 0x53545250 = ACSII(PRTS) */
 CALI_CONFIGURABLE_DEFINE(psnat_len, 0x4c545250) /* be 0x4c545250 = ACSII(PRTL) */
 CALI_CONFIGURABLE_DEFINE(flags, 0x00000001)
-
+CALI_CONFIGURABLE_DEFINE(host_tunnel_ip, 0x4c4e5554) /* be 0x4c4e5554 = ACSII(TUNL) */
```
Comment (mazdakn) [inline felix/bpf-gpl/bpf.h:264]: For my information, `0x4c4e5554 = ACSII(TUNL)` and similar ones are to set the default value?

Comment (mazdakn) [reply felix/bpf-gpl/bpf.h:264]: Ah, got it. It is used by `PatchHostTunnelIPv4()`

**Review by mazdakn — APPROVED** [review]
LGTM

**Review by sridhartigera — APPROVED** [review]

File: felix/bpf-gpl/bpf.h (hunk 1 above)
Comment (tomastigera) [reply felix/bpf-gpl/bpf.h:264]: yup! Just a magic number that se search for, hoping it does not clash with anything else :sweat_smile: It is a relict for ut tests, the libbpf configuration is much better and safer.

**Pull request merged**